common/src/main/java/org/dash/wallet/common/ui/receive/ReceiveInfoView.kt

```diff
@@ -0,0 +1,193 @@
+    private var amount: Coin? = null
+    private var paymentRequestUri: String = ""
+    private var username: String? = null
+
+    init {
+        binding = ReceiveInfoViewBinding.inflate(LayoutInflater.from(context), this)
+        val attrsArray = context.obtainStyledAttributes(attrs, R.styleable.ReceiveInfoView)
+
+        try {
+            val showAmountAction = attrsArray.getBoolean(R.styleable.ReceiveInfoView_ri_show_amount_action, true)
+            binding.specifyAmountButton.isVisible = showAmountAction
+            val showShareAction = attrsArray.getBoolean(R.styleable.ReceiveInfoView_ri_show_share_action, true)
+            binding.shareButton.isVisible = showShareAction
+
+            val qrPreviewScale = attrsArray.getFloat(R.styleable.ReceiveInfoView_ri_qr_code_scale, 1.0f)
+            (binding.qrPreviewBg.layoutParams as LayoutParams).matchConstraintPercentWidth = qrPreviewScale
+            (binding.qrDashLogo.layoutParams as LayoutParams).matchConstraintPercentWidth = (qrPreviewScale / 4.8f)
+        } finally {
+            attrsArray.recycle()
+        }
+
+        if (!isInEditMode) {
+            binding.addressCopyBtn.setOnClickListener {
+                onAddressClicked?.invoke()
+                address?.let { handleCopyAddress(it) }
+            }
+            binding.usernameCopyBtn.setOnClickListener {
+                username?.let { handleCopyUsername(it) }
+            }
+            binding.specifyAmountButton.setOnClickListener {
+                onSpecifyAmountClicked?.invoke()
+            }
+            binding.shareButton.setOnClickListener {
+                onShareClicked?.invoke()
+                handleShare(paymentRequestUri)
+            }
+
+            refresh()
+        }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            binding.qrPreviewBg.isForceDarkAllowed = false
+        }
+    }
+
+    fun setInfo(address: Address, amount: Coin?) {
+        if (this.address != address) {
+            this.address = address
+            this.amount = amount
+            refresh()
+        } else if (this.amount != amount) {
+            this.amount = amount
+            refreshQr()
+        }
+    }
+
+    fun setProfile(username: String?, displayName: String?, avatar: String?, avatarHash: ByteArray?) {
+        this.username = username
+
+        if (!username.isNullOrEmpty()) {
+            binding.usernamePreviewPane.isVisible = true
+            binding.usernamePreview.text = username
+
+            if (!displayName.isNullOrEmpty()) {
+                binding.usernameLabel.text = displayName
+            } else {
+                binding.usernameLabel.text = context.getString(R.string.username)
+            }
+
+            if (!avatar.isNullOrEmpty()) {
+                ProfilePictureDisplay.display(binding.avatar, avatar, avatarHash, username, false, null)
+            }
+        } else {
+            binding.usernamePreviewPane.isVisible = false
+        }
+    }
```

Comment:
The old `ReceiveInfoView` was removed, new one was extended with username and avatar.